spirv_common.hpp

```diff
@@ -107,7 +107,11 @@ inline std::string convert_to_string(float t)
 	// std::to_string for floating point values is broken.
 	// Fallback to something more sane.
 	char buf[64];
+
+	auto old = std::locale::global(std::locale::classic());
```

Comment:
This seems a bit inefficient to do for every convert_to_string. I'd have this in ::compile(), wrap it in an RAII class and then go back to the existing locale in the destructor.